Content.Server/PneumaticCannon/PneumaticCannonSystem.cs

```diff
@@ -0,0 +1,274 @@
+
+                }
+                foreach (var entity in ents)
+                {
+                    storage.Remove(entity);
+                    var dir = (click.ToMapPos(EntityManager) - user.Transform.WorldPosition).Normalized;
+
+                    var randomAngle = GetRandomFireAngleFromPower(component.Power).RotateVec(dir);
+                    var randomStrengthMult = _random.NextFloat(0.75f, 1.25f);
+                    var throwMult = GetRangeMultFromPower(component.Power);
+
+                    var data = new FireData
+                    {
+                        User = user,
+                        Item = entity,
+                        Strength = component.ThrowStrength * randomStrengthMult,
+                        Sound = component.FireSound,
+                        Direction = (dir + randomAngle).Normalized * component.BaseThrowRange * throwMult,
+                    };
+                    _fireQueue.Enqueue(data);
+                }
+
+                if (component.Power == PneumaticCannonPower.High)
+                {
+                    // if power was high, fall on our ass
+                    if (user.TryGetComponent<StunnableComponent>(out var stun))
+                    {
+                        stun.Paralyze(3);
+                        user.PopupMessage(Loc.GetString("pneumatic-cannon-component-power-stun",
+                            ("cannon", component.Owner)));
+                    }
+                }
+            }
+        }
+
+        public void TryRemoveGasTank(PneumaticCannonComponent component, IEntity user)
+        {
+            if (component.GasTankSlot.ContainedEntity == null)
+            {
+                user.PopupMessage(Loc.GetString("pneumatic-cannon-component-gas-tank-none",
+                    ("cannon", component.Owner)));
+                return;
+            }
+
+            var ent = component.GasTankSlot.ContainedEntity;
+            if (component.GasTankSlot.Remove(ent))
+            {
+                if (user.TryGetComponent<HandsComponent>(out var hands))
+                {
+                    hands.TryPutInActiveHandOrAny(ent);
+                }
+
+                user.PopupMessage(Loc.GetString("pneumatic-cannon-component-gas-tank-remove",
+                    ("tank", ent), ("cannon", component.Owner)));
+                UpdateAppearance(component);
+            }
+        }
+
+        public void TryEjectAllItems(PneumaticCannonComponent component, IEntity user)
+        {
+            if (component.Owner.TryGetComponent<ServerStorageComponent>(out var storage))
+            {
+                if (storage.StoredEntities == null) return;
+                foreach (var entity in storage.StoredEntities.ToArray())
+                {
+                    storage.Remove(entity);
+                    if (user.TryGetComponent<HandsComponent>(out var hands))
+                    {
+                        hands.TryPutInActiveHandOrAny(entity);
+                    }
```

Comment:
It sounds like me that eject items should just dump everything onto the ground instead of actually putting anything in the user's hand. I'm imagining I'm dumping a bunch of trash on the ground and also shaking it to dump it out here.